tools/src/h5repack/h5repack_main.c

```diff
@@ -924,6 +933,32 @@ parse_command_line(int argc, const char *const *argv, pack_opt_t *options)
 
         options->fin_fapl = tmp_fapl;
     }
+    if (page_cache > 0) {
+        if ((tmp_fapl = h5tools_get_fapl(options->fin_fapl, NULL, NULL)) < 0) {
+            error_msg("failed to setup FAPL for input file\n");
+            h5tools_setstatus(EXIT_FAILURE);
+            ret_value = -1;
+            goto done;
+        }
+
+        if (H5Pset_page_buffer_size(tmp_fapl, page_cache, 0, 0) < 0) {
+            error_msg("unable to set page buffer cache size for file access\n");
+            h5tools_setstatus(EXIT_FAILURE);
+            ret_value = -1;
+            goto done;
+        }
+
+        /* Close old FAPL */
+        if (options->fin_fapl != H5P_DEFAULT)
+            if (H5Pclose(options->fin_fapl) < 0) {
+                error_msg("failed to close FAPL\n");
+                h5tools_setstatus(EXIT_FAILURE);
+                ret_value = -1;
+                goto done;
+            }
+
+        options->fin_fapl = tmp_fapl;
```

Comment:
The logic in h5repack_copy.c is also checking `options->fin_fapl` against H5P_DEFAULT since a VOL/VFD used to be the only option that would create a non-default one. That logic may need to be changed as well to something like `custom_in_vol || custom_in_vfd` from above, but these booleans would probably need to be added to the `options` struct.